api/src/models/tournament.js

```diff
@@ -0,0 +1,103 @@
+const mongoose = require("mongoose");
+const ObjectId = mongoose.Types.ObjectId;
+const { enumTournamentStageType, enumTournamentStageStatus, enumTournamentStatus } = require("../enums/enumTournament");
+const { enumNumberOfPlayersForGame } = require("../enums/enumModes");
+const { enumMaps } = require("../enums/enumMaps");
+const { enumModes } = require("../enums/enumModes");
+
+const MODELNAME = "tournament";
+
+const TeamSchema = new mongoose.Schema({
+  name: { type: String, trim: true },
+  players: { type: [ObjectId], ref: "user", default: [] },
+  disqualifiedAt: { type: Date, default: null },
+  registeredAt: { type: Date, default: Date.now },
+});
+
+const StageSchema = new mongoose.Schema({
+  name: { type: String, trim: true },
+  teams: { type: [TeamSchema], default: [] },
+  type: { type: String, enum: enumTournamentStageType },
+  status: { type: String, enum: enumTournamentStageStatus, default: enumTournamentStageStatus.PENDING },
+
+  numberOfPlayersForGame: { type: Number, default: enumNumberOfPlayersForGame.twoVTwo },
+  numberOfPlayersPerTeam: { type: Number, default: 2 },
+  maps: { type: [String], default: [enumMaps.ctf_5, enumMaps.ctf_duskwood, enumMaps.ctf_cryochasm, enumMaps.ctf_mars, enumMaps.ctf_moon] },
+  mode: { type: String, enum: enumModes, default: enumModes.twoVTwo },
+
+  modeId: { type: ObjectId, ref: "mode" },
+  modeName: { type: String, trim: true },
+
+  bestOf: { type: Number, default: 3 },
```

Comment:
Maybe introduce a matchConfig here?

Reply:
That config could hold a name, a number of games and potentially later some rules for how the matches would happen, e.g., whether the players can ban etc.